src/System.Net.Security/src/System/Net/SecureProtocols/NegotiateStream.cs

```diff
@@ -0,0 +1,646 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Security.Authentication.ExtendedProtection;
+using System.Security.Principal;
+
+namespace System.Net.Security
+{
+    /*
+        An authenticated stream based on NEGO SSP.
+
+            The class that can be used by client and server side applications
+            - to transfer Identities across the stream
+            - to encrypt data based on NEGO SSP package
+
+            In most cases the innerStream will be of type NetworkStream.
+            On Win9x data encryption is not available and both sides have
```

Comment:
Left this in because even Win9x is not a target platform, the implementation must be able to inter-operate with any OS.